Bottom: 7c19adc0aa0a0e56901b6895e401af1fc2008c17
Top:    f57adac07ac64ab61b710c31b73a194695304fb8
Author: Jason Wang <jasowang@redhat.com>
Date:   2016-06-29 15:55:33 +0800

Refresh of ptr_ring-support-zero-length

---

diff --git a/include/linux/ptr_ring.h b/include/linux/ptr_ring.h
index 42aad79..2052011 100644
--- a/include/linux/ptr_ring.h
+++ b/include/linux/ptr_ring.h
@@ -102,7 +102,7 @@ static inline bool ptr_ring_full_bh(struct ptr_ring *r)
  */
 static inline int __ptr_ring_produce(struct ptr_ring *r, void *ptr)
 {
-	if (r->queue[r->producer])
+	if (unlikely(!r->size) || r->queue[r->producer])
 		return -ENOSPC;
 
 	r->queue[r->producer++] = ptr;
